server/src/com/thoughtworks/go/server/messaging/plugin/StageStatusPluginNotifier.java

```diff
@@ -16,49 +16,258 @@
+    private PipelineService pipelineService;
     private PluginNotificationQueue pluginNotificationQueue;
 
     @Autowired
-    public StageStatusPluginNotifier(PluginNotificationQueue pluginNotificationQueue) {
+    public StageStatusPluginNotifier(NotificationPluginRegistry notificationPluginRegistry, PipelineService pipelineService, PluginNotificationQueue pluginNotificationQueue) {
+        this.notificationPluginRegistry = notificationPluginRegistry;
+        this.pipelineService = pipelineService;
         this.pluginNotificationQueue = pluginNotificationQueue;
     }
 
     @Override
     public void stageStatusChanged(final Stage stage) {
-        Map data = createRequestDataMap(stage);
+        if (notificationPluginRegistry.isAnyPluginInterestedIn(NotificationExtension.STAGE_STATUS_CHANGE_NOTIFICATION)) {
+            Map data = createRequestDataMap(stage);
 
-        pluginNotificationQueue.post(new PluginNotificationMessage(NotificationExtension.STAGE_STATUS_CHANGE_NOTIFICATION, data));
+            pluginNotificationQueue.post(new PluginNotificationMessage(NotificationExtension.STAGE_STATUS_CHANGE_NOTIFICATION, data));
+        }
     }
 
     Map createRequestDataMap(Stage stage) {
-        Map data = new LinkedHashMap();
+        Map<String, Object> data = new LinkedHashMap<String, Object>();
         data.put("pipeline-name", stage.getIdentifier().getPipelineName());
-        data.put("pipeline-counter", stage.getIdentifier().getPipelineCounter());
+        data.put("pipeline-counter", new Integer(stage.getIdentifier().getPipelineCounter()).toString());
         data.put("stage-name", stage.getIdentifier().getStageName());
         data.put("stage-counter", stage.getIdentifier().getStageCounter());
-        data.put("stage-state", stage.getState());
-        data.put("stage-result", stage.getResult());
-        data.put("create-time", timestampToString(stage.getCreatedTime()));
-        data.put("last-transition-time", timestampToString(stage.getLastTransitionedTime()));
+        data.put("stage-state", stage.getState().toString());
+        data.put("stage-result", stage.getResult().toString());
+
+        Pipeline pipeline = pipelineService.fullPipelineById(stage.getPipelineId());
+        Map<String, Object> pipelineMap = createPipelineDataMap(pipeline);
+        data.put("pipeline", pipelineMap);
+
         return data;
     }
 
+    private Map<String, Object> createPipelineDataMap(Pipeline pipeline) {
+        Map<String, Object> pipelineMap = new LinkedHashMap<String, Object>();
+        pipelineMap.put("name", pipeline.getName());
+        pipelineMap.put("counter", new Integer(pipeline.getCounter()).toString());
+        pipelineMap.put("label", pipeline.getLabel());
+
+        List<Map> materialRevisionList = createBuildCauseDataMap(pipeline.getMaterialRevisions());
+        pipelineMap.put("build-cause", materialRevisionList);
+
+        List<Map> stageList = new ArrayList<Map>();
```

Comment:
Definitely worth involving notification plugin authors like @matt-richardson (cc'ed now, I suppose) for thoughts about this. Since he is probably using this, he will know best what is useful. We can then talk about what is performant and possible.